stakewise_cli/storages/database.py

```diff
@@ -0,0 +1,174 @@
+                    raise click.ClickException(
+                        f"Public key {public_key} not presented in {self.deposit_data_ipfs_link}"
+                    )
+
+                data = bytes(str(private_key), "ascii")
+                cipher = self.generate_cipher()
+                encrypted_private_key = cipher.encrypt(data)
+                keystores.append(
+                    DatabaseKeyRecord(
+                        public_key=public_key,
+                        private_key=self._bytes_to_str(encrypted_private_key),
+                        nonce=self._bytes_to_str(cipher.nonce),
+                        validator_index=index // self.max_keys_per_validator,
+                    )
+                )
+                index += 1
+
+                bar.update(1)
+        return keystores
+
+    @cached_property
+    def validators_count(self) -> int:
+        return len(self.keys) // self.max_keys_per_validator
+
+    @cached_property
+    def deposit_data_ipfs_link(self) -> str:
+        return get_operator_deposit_data_ipfs_link(
+            self.sw_gql_client, self.operator_address
+        )
+
+    @cached_property
+    def operator_deposit_data_public_keys(self) -> Set[HexStr]:
+        """Returns operator's deposit data public keys."""
+
+        result: Set[HexStr] = set()
+        if not self.deposit_data_ipfs_link:
+            return result
+
+        deposit_datum = ipfs_fetch(self.deposit_data_ipfs_link)
+        for deposit_data in deposit_datum:
+            public_key = deposit_data["public_key"]
+            if public_key in result:
+                raise click.ClickException(
+                    f"Public key {public_key} is presented twice in {self.deposit_data_ipfs_link}"
+                )
+            result.add(public_key)
+
+        return result
+
+    def init_db(self) -> None:
+        conn = get_db_connection(self.db_url)
+        cur = conn.cursor()
+        cur.execute(
+            """
+            DROP TABLE IF EXISTS keys;
+            CREATE TABLE keys (
+                public_key TEXT UNIQUE NOT NULL,
+                private_key TEXT UNIQUE NOT NULL,
+                nonce TEXT NOT NULL,
+                validator_index TEXT NOT NULL)
+            ;"""
+        )
+
+        conn.commit()
+        cur.close()
+        conn.close()
+
+    def save_to_db(self, keys: List[DatabaseKeyRecord]) -> None:
+        conn = get_db_connection(self.db_url)
+        cur = conn.cursor()
+        execute_values(
+            cur,
+            "INSERT INTO keys (public_key, private_key, nonce, validator_index) VALUES %s",
+            [
+                (x["public_key"], x["private_key"], x["nonce"], x["validator_index"])
+                for x in keys
+            ],
+        )
+        conn.commit()
+        cur.close()
+        conn.close()
+
+    def apply_changes(self) -> None:
+        """Updates database records to new state."""
+        self.init_db()
+        self.save_to_db(self.keys)
```

Comment:
We're only storing keys that are located in the deposit data but ignore those that were registered but are not located in the deposit data. Also, self.keys can be used directly in save_to_db